pyproject.toml

```diff
@@ -39,6 +39,30 @@ full = [
     "httpx>=0.27.0,<0.29.0",
 ]
 
+[dependency-groups]
+dev = [
+    # We add starlette[full] so `uv sync` considers the extras.
+    "starlette[full]",
+    "coverage==7.8.2",
+    "importlib-metadata==8.7.0",
+    "mypy==1.16.1",
+    "ruff==0.12.1",
+    "types-PyYAML==6.0.12.20250516",
+    "pytest==8.4.1",
+    "trio==0.30.0",
+    # Check dist
+    "twine==6.1.0"
```

Comment:
Since we have a lockfile, we can just restrict the minimum here. I'll do this after.